src/jdk.internal.md/share/classes/module-info.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+// This module is primarily an import from a recent tagged version of
+//     https://github.com/commonmark/commonmark-java
+//
+// The following parts are imported:
+//
+// * commonmark/src/main/java
+// * commonmark/src/main/resources
+// * commonmark-ext-gfm-tables/src/main/java
+// * commonmark-ext-gfm-tables/src/main/resources
+//
+// For source and resource files, the following transformations are applied:
+//
+// * legal headers are added
+// * package and import statements are updated
+// * characters outside the ASCII range are converted to Unicode escapes
+// * @SuppressWarnings("fallthrough") is added to getSetextHeadingLevel
```

Comment:
It's possible, but that would be a more global setting, whereas this is a very targeted modification.

Reply:
> I guess it depends how much we want to import the code as-is, without knowing any lint-warts.

I think it would be nice not to have to modify code beyond superficial edits: addition of headers, renaming of packages, and converting of non-ASCII symbols.

Reply:
I think adding `@SuppressWarnings` is a superficial edit, but I see the writing on the wall.

Reply:
Based on additional private conversations, and because the annotation is added automatically by the same utility used to address other issues in the imported code (package declarations, import statements, non-ASCII characters, etc), I will leave the annotation in at this time.